SConstruct

```diff
@@ -1430,6 +1430,13 @@ env['LIBM'] = ['m'] if env['NEED_LIBM'] else []
 
 
 if env['system_sundials'] in ("y", "default"):
+    retcode, uses_mpi = run_preprocessor(
+        conf, ['"sundials/sundials_config.h"'], "SUNDIALS_MPI_ENABLED")
+    if retcode and uses_mpi == "1":
+        if not conf.CheckCXXHeader('mpi.h', '""'):
+            logger.error("System Sundials was compiled with MPI support, but header "
+                         "'mpi.h' was not found.")
+            sys.exit(1)
```

Comment:
I think it's worth suggesting the two possible resolutions here; most users won't know what to do with "'mpi.h' was not found". Namely:
- add `CC=mpicc CXX=mpicxx` to the `scons build` command line
- Use the bundled version of SUNDIALS by specifying the `system_sundials=n` option to SCons.

Reply:
Updated. Now we have the following:
```
ERROR: System installation of Sundials version 7.1.1 has MPI support enabled,
but the required 'mpi.h' header file cannot be found. To resolve this issue,
choose one of the following SCons options when running the 'scons build' command:
- Add 'CC=mpicc CXX=mpicxx' to use MPI compiler wrappers, or
- Add 'system_sundials=n' to disable the system Sundials installation.
```

Reply:
PS: had to fix a regression as the check only applies to Sundials>=7.0.0. Before that, there aren't any issues.